### DISCUSSION TITLE
Running preview version is not straight forward

### DISCUSSION THREAD

**Discussion by tomkerkhove** [arcus-azure/arcus.observability#211] [Q&A] [answered]
See @meersschautarnaud's  remark - https://github.com/arcus-azure/arcus.observability/issues/209#issuecomment-831438577 and @fgheysels had issues as well which were sent offline.

**Comment by fgheysels** [comment, marked as answer]
This is strange though, since the transitive dependencies are also updated when you reference the MyGet build:

![image](https://user-images.githubusercontent.com/3605786/117003076-affc4a80-ace4-11eb-968c-713f1e97cbda.png)

Wait, Arcus.WebApi.Logging.Core is not.

**Reply by fgheysels** [reply]
Is it because the official build pipeline sets an assembly-version in the assembly, and the MyGet version does not ?

I'm pretty sure it is.

When I look at the manifest of `Arcus.WebApi.Logging', I can see that the version of Arcus.Observability.Correlation assembly that it references is 2.0.0

![image](https://user-images.githubusercontent.com/3605786/117006771-1e430c00-ace9-11eb-8412-5f23399b3062.png)

Offcourse, the version of Arcus.Observability.Correlation that comes with the preview built, has set its assembly version to 1.0.0
![image](https://user-images.githubusercontent.com/3605786/117006938-50ed0480-ace9-11eb-9544-2aa96c26b89f.png)
This version is lower then the version that is referenced in the manifest, and hence, the assembly-loading fails.

**Reply by fgheysels** [reply]
Is it possible to provide a MyGet build where the assembly version of the Arcus.Correlation assemblies is defined and set to a version that is higher then 2.0.0 ?

**Reply by tomkerkhove** [reply]
I think we should change our MyGet packages to target vNext as an assembly version maybe?

**Reply by fgheysels** [reply]
Could you elaborate a little bit on that ?

**Reply by tomkerkhove** [reply]
If we release Arcus Observability v2.2, our MyGet packages should be 2.3-something so that it's a higher version.

This would be in combination with https://github.com/arcus-azure/arcus/issues/129 where we don't enforce a specific version on you.

**Reply by fgheysels** [reply]
We can try, but I'm not convinced that it'll work, since this error is coming from an assembly level.
Assembly A from Arcus.Project1 references assembly B with version 2.0 from Arcus.Project2.  This link is created at build time.   So, I don't know if it'll work, but we should give it a try

**Reply by fgheysels** [reply]
Tried the preview version (20210505.2.0-preview).

In my API project, I still run into the AssemblyLoadException
![image](https://user-images.githubusercontent.com/3605786/117172808-6f77fc00-adcc-11eb-9c22-2e93ac5270df.png)

That is because the assembly version of the preview build (Arcus.Observability.Correlation) is set to 1.0.0.0:
![image](https://user-images.githubusercontent.com/3605786/117173081-b36b0100-adcc-11eb-8822-f0c372318e02.png)

while Arcus.WebApi.Logging still references these versions:

![image](https://user-images.githubusercontent.com/3605786/117173390-ff1daa80-adcc-11eb-8907-9ba69adb9509.png)

**Reply by fgheysels** [reply]
mea culpa maximus.

I had used a MyGet preview version (20210505.2.0-preview) instead of the preview version on nuget (2.2.0-preview-1).
With 2.2.0-preview-1, the assemblyload exception does not occur anymore.

**Reply by fgheysels** [reply]
But, with the 2.2.0-preview 1 build, my ConsoleLogger still logs the events like this:

```json
{ "ts": 2021-05-06T14:35:48.7273716+02:00, "level": "Warning",   "message": "Events SomeArcusEvent (Context: {"Item": "foo", "Error": "bar"})", "properties": {"SourceContext": "ArcusObservabilityConsoleTest.TestClass", "Item": "foo", "Error": "bar"} }
```
where it should rather be something like this:

```
{ "ts": 2021-05-06T14:31:02.1215702+02:00, "level": "Warning",   "message": "{"Type": "Event", "Message": "SomeEvent", "Properties": {"Item": "foo", "Error": "bar"}, "$type": "LogEntry"}", "properties": {"SourceContext": "ArcusObservabilityConsoleTest.TestClass"} }
```
(The above is done by writting the event like this:

```csharp
var scope = new Dictionary<string, object>()
{
    ["Item"] = "foo",
    ["Error"] = "bar"
};

LogEntry entry = new LogEntry()
{
    Type = "Event",
    Message = "SomeEvent",
    Properties = scope
};

_logger.LogWarning("{@entry}", entry);
```

**Reply by tomkerkhove** [reply]
Let's move that conversation to the issue and keep this one to track the preview version pains

**Reply by fgheysels** [reply]
True.  Added some additional info in the [issue](https://github.com/arcus-azure/arcus.observability/issues/205#issuecomment-833453920).

**Reply by stijnmoreels** [reply]
This is picked up here https://github.com/arcus-azure/azure-devops-templates/issues/45
This will make sure that future preview releases will start with the major-minor-patch version of the current package version - which would make combinations of preview/full packages easier. Especially since we also invested in using NuGet package ranges throughout our libraries.
